internal/integration/api/extensions_nvidia.go

```diff
@@ -0,0 +1,307 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+//go:build integration_api
+
+package api
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"github.com/siderolabs/go-pointer"
+	"github.com/siderolabs/go-retry/retry"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	nodev1 "k8s.io/api/node/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/siderolabs/talos/internal/integration/base"
+)
+
+// ExtensionsSuiteNVIDIA verifies Talos is securebooted.
+type ExtensionsSuiteNVIDIA struct {
+	base.K8sSuite
+
+	ctx       context.Context //nolint:containedctx
+	ctxCancel context.CancelFunc
+}
+
+// SuiteName ...
+func (suite *ExtensionsSuiteNVIDIA) SuiteName() string {
+	return "api.ExtensionsSuiteNVIDIA"
+}
+
+// SetupTest ...
+func (suite *ExtensionsSuiteNVIDIA) SetupTest() {
+	if !suite.ExtensionsNvidia {
+		suite.T().Skip("skipping as nvidia extensions test are not enabled")
+	}
+
+	// make sure API calls have timeout
+	suite.ctx, suite.ctxCancel = context.WithTimeout(context.Background(), 5*time.Minute)
+}
+
+// TearDownTest ...
+func (suite *ExtensionsSuiteNVIDIA) TearDownTest() {
+	if suite.ctxCancel != nil {
+		suite.ctxCancel()
+	}
+}
+
+// TestExtensionsNVIDIA verifies that a cuda workload can be run.
+func (suite *ExtensionsSuiteNVIDIA) TestExtensionsNVIDIA() {
+	expectedModulesModDep := map[string]string{
+		"nvidia":         "nvidia.ko",
+		"nvidia_uvm":     "nvidia-uvm.ko",
+		"nvidia_drm":     "nvidia-drm.ko",
+		"nvidia_modeset": "nvidia-modeset.ko",
+	}
+
+	// if we're testing NVIDIA stuff we need to get the nodes having NVIDIA GPUs
+	// we query k8s to get the nodes having the label node.kubernetes.io/instance-type.
+	// this label is set by the cloud provider and it's value is the instance type.
+	// the nvidia e2e-aws tests creates gpu nodes one with g4dn.xlarge and another
+	// with p4d.24xlarge
+	for _, nvidiaNode := range suite.getNVIDIANodes("node.kubernetes.io/instance-type in (g4dn.xlarge, p4d.24xlarge)") {
+		suite.AssertExpectedModules(suite.ctx, nvidiaNode, expectedModulesModDep)
+	}
+
+	nodes := suite.getNVIDIANodes("node.kubernetes.io/instance-type=g4dn.xlarge")
+	for _, node := range nodes {
+		suite.AssertServicesRunning(suite.ctx, node, map[string]string{
+			"ext-nvidia-persistenced": "Running",
+		})
+	}
+
+	// nodes = suite.getNVIDIANodes("node.kubernetes.io/instance-type=p4d.24xlarge")
```

Comment:
is it commented out for some reason?

Reply:
yes, the a100 gpu's are so costly compared to aws, so want to do that only in azure 30$/hr vs 3$/hr